From f147fa6d8e787d0ab450c00bdeb2ccd09fbaf679 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Fri, 28 May 2021 01:50:23 +0200
Subject: [PATCH] Don't use the internal glibc header sys/cdefs.h

for file in $(ag -l __BEGIN_DECLS); do
	sed -i "${file}" \
		-e 's/__BEGIN_DECLS/#ifdef __cplusplus\nextern "C" {\n#endif/g' \
		-e 's/__END_DECLS/#ifdef __cplusplus\n}\n#endif/g' \
		-e '/#include <sys\/cdefs.h>/d'
done

https://wiki.musl-libc.org/faq.html#Q:-When-compiling-something-against-musl,-I-get-error-messages-about
---
 .../include/android/file_descriptor_jni.h                | 9 ++++++---
 tests/lib/testUtil/include/testUtil.h                    | 8 ++++++--
 2 files changed, 12 insertions(+), 5 deletions(-)

diff --git a/module_ndk_libs/libnativehelper/include/android/file_descriptor_jni.h b/module_ndk_libs/libnativehelper/include/android/file_descriptor_jni.h
index 793f7bcf..775734ed 100644
--- a/module_ndk_libs/libnativehelper/include/android/file_descriptor_jni.h
+++ b/module_ndk_libs/libnativehelper/include/android/file_descriptor_jni.h
@@ -25,11 +25,12 @@
 
 #pragma once
 
-#include <sys/cdefs.h>
 
 #include <jni.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 /**
  * Returns a new java.io.FileDescriptor.
@@ -79,6 +80,8 @@ int AFileDescriptor_getFD(JNIEnv* env, jobject fileDescriptor) __INTRODUCED_IN(3
  */
 void AFileDescriptor_setFD(JNIEnv* env, jobject fileDescriptor, int fd) __INTRODUCED_IN(31);
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 /** @} */
diff --git a/tests/lib/testUtil/include/testUtil.h b/tests/lib/testUtil/include/testUtil.h
index 3b75914d..bdef437d 100644
--- a/tests/lib/testUtil/include/testUtil.h
+++ b/tests/lib/testUtil/include/testUtil.h
@@ -22,7 +22,9 @@
 #include <stdio.h>
 #include <sys/time.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 // Time Utilities
 struct timespec double2ts(double amt);
@@ -64,6 +66,8 @@ uint64_t testXDumpGetOffset(void);
 // Command Execution
 void testExecCmd(const char *cmd);
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif
